### PR TITLE
[TT-6486/TT-9024] fix dynamic and static mTLS not blocking requests when certificate expired

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description
https://tyktech.atlassian.net/browse/TT-6486
Currently, if an API is configured with dynamic mTLS or static mTLS the request is not blocked when the client certificate is expired. This PR adds a fix for that.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5087834610)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 5a82c79a21106deb49b14d5d6791963c7abd1ceb 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5087838078)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 2e12ec38a4c59ee3a8b5558d6efc8dd00bb09765 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5087863468)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 8b364f00291b7f470b3c6b2645095657af30ac88 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5087895961)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: bac2922ac07d1326d3bd7941b987bf73e4bd9acc 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5087908241)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: ee6c67aad5777e475ecadd56d391c6932ceb401e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5088275770)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: d47c46fe9c6bceadbf43d0d24ccb96a5e618cd63 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5088371944)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: a33888738841f541cccce36a4c1edc3f4acf8583 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090522956)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 39ea3e0c76e6c8e965135374afb47a28411b669c 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090567291)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 94c3bb178e61ba917898204d4e4a7b7f2dd74a52 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090831213)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 2b4087a6c543ed4db85e48c22a7f73be44afe9d6 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090885435)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 94c3bb178e61ba917898204d4e4a7b7f2dd74a52 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090916630)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 12ee11a28f8bf0ef5ca40b2ba6b99a15080fe0b2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090929373)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 9417bb7ad64e74fe8309b32b3c3eb3c069b451a8 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5090987063)

File: certs/manager.go [hunk 1]
```diff
@@ -29,22 +28,42 @@ const (
...
+	ListRawPublicKey(keyID string) (out interface{})
+	ListAllIds(prefix string) (out []string)
+	GetRaw(certID string) (string, error)
+	Add(certData []byte, orgID string) (string, error)
+	Delete(certID string, orgID string)
+	CertPool(certIDs []string) *x509.CertPool
+	FlushCache()
+}
+
+type certificateManager struct {
```
Comment (furkansenharputlu) [inline certs/manager.go:53]: Can this be a breaking change for goplugin users?

File: gateway/handler_error.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ const (
 	MsgKeyNotAuthorized                        = "Key not authorised"
 	MsgOauthClientRevoked                      = "Key not authorised. OAuth client access was revoked"
 	MsgKeyNotAuthorizedUnexpectedSigningMethod = "Key not authorized: Unexpected signing method"
+	MsgCertificateExpired                      = "certificate is expired"
```
Comment (letzya) [inline gateway/handler_error.go:36]: maybe `"certificate has expired"?

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5062/merge
Commit: e3da43e0f776f724b454724de8c7326e6b8fa1b7 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5119253205)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5062/merge
Commit: 7f09a5a0e2fc30977c2e94688fffcb91ed5179be 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5120376634)

**Comment by Tyk-ITS** [comment]
**API tests result: cancelled** :no_entry_sign: 
Branch used: refs/pull/5062/merge
Commit: 63da80a43f9c3fb86e76e5195cac73d85d921b25 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5117945368)

**Pull request merged**